client/beefy/src/communication/request_response/outgoing_requests_engine.rs

```diff
@@ -43,27 +45,41 @@ type Response = Result<Vec<u8>, RequestFailure>;
 type ResponseReceiver = oneshot::Receiver<Response>;
 
 #[derive(Clone, Debug)]
-struct RequestInfo<B: Block> {
+struct RequestInfo<B: Block, AuthId: Encode + Decode + Debug + Clone + Ord + Sync + Send> {
 	block: NumberFor<B>,
-	active_set: ValidatorSet<AuthorityId>,
+	active_set: ValidatorSet<AuthId>,
 }
 
-enum State<B: Block> {
+enum State<B: Block, AuthId: Encode + Decode + Debug + Clone + Ord + Sync + Send> {
 	Idle,
-	AwaitingResponse(PeerId, RequestInfo<B>, ResponseReceiver),
+	AwaitingResponse(PeerId, RequestInfo<B, AuthId>, ResponseReceiver),
 }
 
-pub struct OnDemandJustificationsEngine<B: Block> {
+pub struct OnDemandJustificationsEngine<
+	B: Block,
+	AuthId: Encode + Decode + Debug + Clone + Ord + Sync + Send,
+	TSignature: Encode + Decode + Debug + Clone + Sync + Send,
+	BKS: BeefyKeystore<AuthId, TSignature, Public = AuthId>,
+> {
 	network: Arc<dyn NetworkRequest + Send + Sync>,
 	protocol_name: ProtocolName,
 
 	live_peers: Arc<Mutex<KnownPeers<B>>>,
 	peers_cache: VecDeque<PeerId>,
 
-	state: State<B>,
+	state: State<B, AuthId>,
+	_auth_id: PhantomData<AuthId>,
+	_signature: PhantomData<TSignature>,
+	_keystor: PhantomData<BKS>,
```

Comment:
I saw a couple of places where we have fields declared like this. Do we plan to use them in the future or is it just a way of declaring phantom data?

If we plan to use them in the future, I think it would be helpful to write in the description of the PR what is the current status, what remains to be done, and maybe also mark the PR as a draft if needed.